src/nvim/api/vim.c

```diff
@@ -54,6 +54,92 @@ void nvim_command(String command, Error *err)
   try_end(err);
 }
 
+/// Retrieves highlight description from its name
+///
+/// @param name Highlight group name
+/// @return a highlight description in dictionary {'hl_id': X, 'bg': Y, 'fg': Z}
+/// @see nvim_hl_from_id
+Dictionary nvim_hl_from_name(String name, Error *err)
+  FUNC_API_SINCE(3)
+{
+  Dictionary result = ARRAY_DICT_INIT;
+  int id = syn_name2id((const char_u *)name.data);
+
+  if (hl_invalid_id(id)) {
+    api_set_error(err, kErrorTypeException, "Invalid highlight name %s",
+                  name.data);
+    return result;
+  }
+  result = nvim_hl_from_id(id, err);
+  return result;
+}
+
+/// Retrieves highlight description from its id
+///
+/// @param name Highlight group name
+/// @return a highlight description in a dictionary
+/// @see nvim_hl_from_id
+Dictionary nvim_hl_from_id(Integer hl_id, Error *err)
+  FUNC_API_SINCE(3)
+{
+  HlAttrs attrs;
+  Dictionary dic = ARRAY_DICT_INIT;
+
+  PUT(dic, "hl_id", INTEGER_OBJ(hl_id));
+
+  int attrcode = syn_id2attr((int)hl_id);
+  bool res = attr2hlattr(attrcode, true, &attrs);
+
+  // ILOG("found attr=%d for hl_id=%d res=%d", attrcode, hl_id, res);
+  if (hl_invalid_id((int)hl_id)) {
+    api_set_error(err, kErrorTypeException, "Invalid highlight id %d", hl_id);
+    return dic;
+  }
+  /// NEWVERSION
+  // attr = hl2attr (hl_id);
+  // bool res = attr2hlattr (&attr, true, &attrs);
+  // bool res = attrentry2hlattr (attr, true, &attrs);
+  /// NEWVERSION
+  if (res) {
+    dic = attr2dic(attrs);
+  }
+  return (dic);
+}
+
+/// Returns an array of dictionaries as produced by nvim_hl_from_id
+///
+/// @param highlights a list of highlights id (string or integer) to look for
+/// todo return array instead ? ArrayOf(HlAttrs)
+Object nvim_hl_get_list(Array highlights, Error *err)
```

Comment:
if this is a batched version of the methods above then it is not needed (use call_atomic or lua)